cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -856,11 +856,15 @@ private predicate unary_simple_comparison_eq(
     inNonZeroCase = false
   )
   or
-  // There's no implicit CompareInstruction in files compiled as C since C
-  // doesn't have implicit boolean conversions. So instead we check whether
-  // there's a branch on a value of pointer or integer type.
-  relevantUnaryComparison(test) and
```

Comment:
That whole predicate can be removed now, as the only remaining use is a recursion.

Reply:
Oh, right. Well spotted! Fixed in 7f62085be57ef80678323c1de5e964ea6a5afbed